src/attachments/modifyImage.js

```diff
@@ -71,66 +77,101 @@ export function modifyImages(
                   error: false,
                   message: `${doing} image${multiple ? "s" : ""} ${_.map(chunkFiles, file => file.fromPath)
                                                                     .join(", ")}`,
-                  currentStep: status.currentStep,
+                  currentStep,
                   steps
                 });
 
                 const filesInChunkWithStatus = _.map(chunkFiles, file => {
-                  const to = file.toPath;
-                  if (database.find(path.basename(to), key)) {
+                  const {fromPath, toPath} = file;
+
+                  if (database.find(path.basename(toPath), key)) {
                     return {
                       done: true,
-                      fromPath: file.fromPath,
-                      toPath: to
+                      fromPath: fromPath,
+                      toPath: toPath
                     };
                   } else {
                     return {
                       done: false,
-                      fromPath: file.fromPath,
-                      toPath: to
+                      fromPath: fromPath,
+                      toPath: toPath
                     };
                   }
                 });
 
                 return Promise
                   .all(_.map(filesInChunkWithStatus, file => {
                     if (file.done) {
+                      progress({
+                        error: false,
+                        message: `Already modified image ${file.toPath}`,
+                        currentStep,
+                        steps
+                      });
+
                       return Promise.resolve(file);
-                    } else if (resize) { // resize (and maybe minify)
+                    }
+
+                    const handleError = (error) => {
+                      progress({
+                        error: error,
+                        message: `Could not modify image ${file.fromPath}`,
+                        currentStep,
+                        steps
+                      });
+
+                      _.attempt(onError, error, file.fromPath);
+
+                      return null;
```

Comment:
I'd reuse `file` here and add an `error` to it, instead of filtering falsey values with `_.compact` - this way you add information down the pipe and can still decide, what you want to do with it later. I'd try to avoid the removal of information / data if possible.